mobile/library/cc/engine_builder.cc

```diff
@@ -407,8 +414,8 @@ std::string EngineBuilder::generateConfigStr() const {
     insertCustomFilter(filter_config, config_template);
   }
 
-  if (!rtds_layer_name_.empty() && ads_api_type_.empty()) {
-    throw std::runtime_error("ADS must be configured when using RTDS");
+  if ((!rtds_layer_name_.empty() || enable_cds_) && ads_api_type_.empty()) {
+    throw std::runtime_error("ADS must be configured when using XDS");
```

Comment:
nit: s/XDS/xDS

Reply:
Question: did we decide if we want to make this an EnvoyException instead of runtime_error?

Reply:
I don't remember but it seems safe enough to follow the convention of this file.